engine/src/elections.rs

```diff
@@ -254,21 +261,10 @@ where
 
 						for (unprovided_shared_data_hash, reference_details) in electoral_data.unprovided_shared_data_hashes {
 							if let Some((shared_data, _)) = shared_data_cache.get(&unprovided_shared_data_hash) {
-								if (reference_details.created..reference_details.expires).contains(&block_info.number) {
-									// Increase probability until expiry
-									let lerp_factor = ((block_info.number - reference_details.created + 1) as f64) / ((reference_details.expires - reference_details.created) as f64);
-
-									// Starting with a low probability avoids problems were everyone has in-flight votes (i.e. has the shared data), but only a few validators have votes in_blocks. Ideally we would only try to submit shared data if one of the associated votes we made was "in_block".
-									let initial_probability = 1.0 / (core::cmp::max(1, electoral_data.authority_count) as f64);
-
-									// `Vote`s should not contain the same `SharedData` value repeatedly as this will make this under-estimate the probability each engine should use when providing that SharedData, as `reference_details.count` will not be an accurate estimate of the number of authorities that have this SharedData and therefore are going to try and submit it.
-									let final_probability = 1.0 / (core::cmp::max(1, core::cmp::min(reference_details.count, electoral_data.authority_count)) as f64);
-
-									if rng.gen_bool((1.0 - lerp_factor) * initial_probability + lerp_factor * final_probability) {
-										self.state_chain_client.submit_signed_extrinsic(pallet_cf_elections::Call::<state_chain_runtime::Runtime, Instance>::provide_shared_data {
-											shared_data: shared_data.clone(),
-										}).await;
-									}
+								if (reference_details.created..reference_details.expires).contains(&block_info.number) && rng.gen_bool(required_full_vote_probability(authority_count, 0.01)) {
```

Comment:
Would also include a small comment re: what we discussed about the probability of a 2nd block failure